library/ssl_srv.c

```diff
@@ -1785,7 +1785,7 @@ static int ssl_parse_client_hello( mbedtls_ssl_context *ssl )
                                             "during renegotiation" ) );
                 mbedtls_ssl_send_alert_message( ssl, MBEDTLS_SSL_ALERT_LEVEL_FATAL,
                                                 MBEDTLS_SSL_ALERT_MSG_HANDSHAKE_FAILURE );
-                return( MBEDTLS_ERR_SSL_BAD_HS_CLIENT_HELLO );
+                return( MBEDTLS_ERR_SSL_ILLEGAL_PARAMETER );
```

Comment:
Why is the error code not aligned with the alert message here ? Thus why not MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE here ?